doc/internal/man3/ossl_namemap_new.pod

```diff
@@ -25,9 +25,9 @@ ossl_namemap_doall_names
  int ossl_namemap_name2num(const OSSL_NAMEMAP *namemap, const char *name);
  int ossl_namemap_name2num_n(const OSSL_NAMEMAP *namemap,
                              const char *name, size_t name_len);
- void ossl_namemap_doall_names(const OSSL_NAMEMAP *namemap, int number,
-                               void (*fn)(const char *name, void *data),
-                               void *data);
+ int ossl_namemap_doall_names(const OSSL_NAMEMAP *namemap, int number,
```

Comment:
I think a note to the effect of: _if this function returns error, the callback will not have been called_ would be beneficial for this one.

I.e. provide a guarantee that either zero or all names are sent to the callback function.


Doing the same for the EVP wrappers might be worth considering too.

Reply:
Fixed